traversal/selector/exploreAll.go

```diff
@@ -0,0 +1,44 @@
+package selector
+
+import (
+	"fmt"
+
+	ipld "github.com/ipld/go-ipld-prime"
+)
+
+// ExploreAll is similar to a `*` -- it traverses all elements of an array,
+// or all entries in a map, and applies a next selector to the reached nodes.
+type ExploreAll struct {
+	next Selector // selector for element we're interested in
+}
+
+// Interests for ExploreAll is empty (meaning traverse everything)
```

Comment:
This is a bit nitty of me, but I like to be very precise about saying "nil" and "zero" (or "empty" for lists and maps, I suppose, but then meaning "not nil").  This one being nil, of course.